front/components/agent_builder/triggers/AgentBuilderTriggersBlock.tsx

```diff
@@ -152,8 +175,32 @@ export function AgentBuilderTriggersBlock({
       {/* Create/Edit Schedule Modal */}
       <ScheduleEditionModal
         owner={owner}
-        trigger={dialogMode?.type === "edit" ? dialogMode.trigger : undefined}
-        isOpen={dialogMode !== null}
+        trigger={
+          dialogMode?.type === "edit" && dialogMode.trigger.kind === "schedule"
+            ? dialogMode.trigger
+            : undefined
```

Comment:
I had to check the DialogMode type to verify - worth creating a few variables just for the sake of readability :
- isEditingSchedule 
- isAddingSchedule
- isEditingTrigger
- isAddingTrigger